plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -392,16 +440,86 @@ def _remove_manifest(cls, repo, manifest):
         cls._purge_unlinked_tags(repo, manifest)
         cls._purge_unlinked_blobs(repo, manifest)
 
+    @classmethod
+    def _remove_manifest_list(cls, repo, manifest_list):
+        """
+        Purge Tags and image manifests associated with a given ManifestList in the repository.
+
+        :param repo:     The affected repository.
+        :type  repo:     pulp.server.db.model.Repository
+        :param manifest_list: The ManifestList being removed
+        :type  manifest_list: pulp_docker.plugins.models.ManifestList
+        """
+        cls._purge_unlinked_tags(repo, manifest_list)
+        cls._purge_unlinked_manifests(repo, manifest_list)
+
+    @staticmethod
+    def _purge_unlinked_manifests(repo, manifest_list):
+
+        # Find manifest digests referenced by removed manifest lists (orphaned)
+        orphaned = set()
+        for image_man in manifest_list.manifests:
+            orphaned.add(image_man)
+            if manifest_list.amd64_digest:
+                orphaned.add(manifest_list.amd64_digest)
```

Comment:
Same question here. Shouldn't ``manifest_list.manifests`` contain all digests, including the one that might appear in ``manifest_list.amd64_digest`` ?